mobilitypath_publisher/src/mobilitypath_publisher.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+#include "mobilitypath_publisher.h"
+
+namespace mobilitypath_publisher
+{
+    
+    MobilityPathPublication::MobilityPathPublication():
+                            x_threshold(20.0),
+                            y_threshold(3.0) {}
+
+    void MobilityPathPublication::initialize()
+    {
+        nh_.reset(new ros::CARMANodeHandle());
+        pnh_.reset(new ros::CARMANodeHandle("~"));
+        pnh_->param<double>("trajectory_threshold_x", x_threshold, 20.0);
+        pnh_->param<double>("trajectory_threshold_y", y_threshold, 3.0);
+        mob_path_pub_ = nh_->advertise<cav_msgs::MobilityPath>("mobility_path_msg", 5);
+        traj_sub_ = nh_->subscribe("plan_trajectory", 5, &MobilityPathPublication::trajectory_cb, this);
+        pose_sub_ = nh_->subscribe("current_pose", 5, &MobilityPathPublication::currentpose_cb, this);
+        ros::CARMANodeHandle::setSpinRate(10.0);
+        tf2_listener_.reset(new tf2_ros::TransformListener(tf2_buffer_));
+    }
+
+    void MobilityPathPublication::run()
+    {
+        initialize();
+        ros::CARMANodeHandle::spin();
+    }
+
+    void MobilityPathPublication::trajectory_cb(const cav_msgs::TrajectoryPlanConstPtr& msg)
+    {
+        latest_trajectory = *msg;
+        cav_msgs::MobilityPath mobility_path = mobilityPathMessageGenerator(latest_trajectory);
+        mob_path_pub_.publish(mobility_path);
+    }
+
+    void MobilityPathPublication::currentpose_cb(const geometry_msgs::PoseStampedConstPtr& msg)
+    {
+        current_pose_ = msg;
+    }
+
+    cav_msgs::MobilityPath MobilityPathPublication::mobilityPathMessageGenerator(cav_msgs::TrajectoryPlan trajectory_plan)
+    {
+        cav_msgs::MobilityPath mobility_path_msg;
+        mobility_path_msg.header = composeMobilityHeader();
+        
+
+        std::vector<cav_msgs::TrajectoryPlanPoint> traj_points;
+        traj_points = trajectory_plan.trajectory_points;
```

Comment:
why is this intermediate variable needed? Seems like you could just pass trajectory_plan.trajectory_points into the function.